lib/readelf.c

```diff
@@ -153,30 +163,46 @@ static bool _is_elf_type(const char *path, int type)
  * Return true if a specified file is an ELF shared library file, that
  * is, of type ET_DYN.
  */
-bool is_elf_shared_library(const char *path)
+bool is_elf_shared_library(rpmfile_entry_t *file)
 {
-    return _is_elf_type(path, ET_DYN);
+    /* -1 "no", 1 "yes", 0 "don't know yet" */
+    if (file->is_elf_shared_library != 0)
```

Comment:
Please use { } around the block of this if statement even if it is only a single line.

Reply:
done